src/main/java/com/google/sps/servlets/URLTest.java

```diff
@@ -0,0 +1,185 @@
+
+   Description:
+      A simple Java program that connects to NyTimes' Covid CSV file
+      then processes that data into an easy to use HashMap data structure.
+      Then, it asks a user which State and County it would like to see
+      Covid data on, ending by returning the desired data.
+      
+      * Prototype for IsCovidNearMe.java *
+      
+   Notes:
+      While the csv could be locally downloaded and ran instead, this particular
+      file is updated on a daily basis- making reading directly from the internet easier to manage.
+      
+      The HashMap setup is scalable, however is setup the way it is using pairs right now because we only
+      have two fields to save per county. If we can get more, (Population, Vaccinations), then instead of 
+      a Pair, we can use custom classes (class County) to store the data in.
+      
+      That being said, the current hashmap uses two "Pairs," or an object that can store two variables in one.
+      
+      To access hashmap:
+         dataMap.get( new Pair<String, String>("myState", "myCounty") );  (returns a pair- cases and deaths)
+         
+      To use pair:
+         myPair.getKey()   (Gets first variable in pair)
+         myPair.getValue() (Gets second variable in pair)
+      
+      
+   Made by rivejona@kean.edu (Jon)
+   
+*/
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Scanner;
+
+import java.net.URL;
+import java.net.URLConnection;
+
+// import javafx.util.Pair;
+
+
+public class URLTest {
+   public static void main(String[] args) throws Exception {
+ 
+// #######################
+// # Retrieving the Data #
+// #######################
+      
+   // Create a Java URL to connect to the CSV we want to read from.
+      URL url = new URL("https://raw.githubusercontent.com/nytimes/covid-19-data/master/live/us-counties.csv");
+      
+   // From that URL, send a "Get" request to retrieve the information.
+      URLConnection data = url.openConnection();
+      
+   // Create a Scanner that can read from our URL response.
+      Scanner input = new Scanner(data.getInputStream());
+      
+      
+// #######################   Notes:
+// # Processing the Data #      Since I want the data to be indexed by State AND County, I must use "Pairs."
+// #######################      Also, since the data we have for every county is also a pair, I'll use one in our data aswell.
+
+   // Create a hashmap to store our data, indexed by a pair of strings, that will store a pair of ints. (State, County) => (cases, deaths)
+      HashMap<Pair<String, String>, Pair<Integer, Integer>> dataMap = new HashMap<Pair<String, String>, Pair<Integer, Integer>>();
+      
+   // Skip the first line of our file, it only contains table headers
+      input.nextLine();
+      
+   // Fill the hashmap with our data! (Read our data line by line, splitting by comma)
+      while(input.hasNextLine()) {
+         String[] tokens = input.nextLine().split(",");
+         
```

Comment:
I think converting the characters in each string for the state and county variables to all upper or lowercase would help if the user inputs any case styling like "oHiO" or "califorNia".

Just adding ".toUpperCase()" or ".toLowerCase()" to the end of the tokens array would solve that.

Reply:
Great idea. Will do!